docs/usage/schema.md

```diff
@@ -695,52 +695,52 @@ class Model(BaseModel):
 
 
 # Default location for OpenAPI
-top_level_schema = models_json_schema([Model], ref_prefix='#/components/schemas/')
+top_level_schema = AnalyzedType.json_schemas(
+    [AnalyzedType(Model)], ref_template='#/components/schemas/{model}'
+)
 print(json.dumps(top_level_schema, indent=2))
+"""
+{
+  "$defs": {
+    "Foo": {
+      "title": "Foo",
+      "type": "object",
+      "properties": {
+        "a": {
+          "type": "integer",
+          "title": "A"
+        }
+      },
+      "required": [
+        "a"
+      ]
+    },
+    "Model": {
+      "title": "Model",
+      "type": "object",
+      "properties": {
+        "a": {
+          "$ref": "#/components/schemas/Foo"
+        }
+      },
+      "required": [
+        "a"
+      ]
+    }
+  }
+}
+"""
 ```
 
-
 It's also possible to extend/override the generated JSON schema in a model.
 
-To do it, use the `Config` sub-class attribute `schema_extra`.
+To do it, implement `__pydantic_modify_json_schema__` on your model:
 
 For example, you could add `examples` to the JSON Schema:
 
-```py output="json" test="xfail - replace schema_extra"
-from pydantic import BaseModel
-
-
-class Person(BaseModel):
-    name: str
-    age: int
-
-    class Config:
-        # TODO: This is no longer valid in v2;
-        #   update example to use __pydantic_modify_json_schema__
-        schema_extra = {
-            'examples': [
-                {
-                    'name': 'John Doe',
-                    'age': 25,
-                }
-            ]
-        }
-
-
-print(Person.schema_json(indent=2))
-```
-
-
-For more fine-grained control, you can alternatively set `schema_extra` to a callable and post-process the generated schema.
-The callable can have one or two positional arguments.
-The first will be the schema dictionary.
-The second, if accepted, will be the model class.
-The callable is expected to mutate the schema dictionary *in-place*; the return value is not used.
```

Comment:
Since it now always has to be a callable having two sections doesn't make sense.